packages/components/pie-webc-testing/src/helpers/page-object/base-page.ts

```diff
@@ -1,10 +1,86 @@
 import { type Page } from '@playwright/test';
+import { buildUrl } from './storybook-extensions';
+
+declare global {
+    interface Window { __eventsArray: Array<string>; }
+}
 
 export class BasePage {
     readonly page: Page;
+    componentName: string;
+    componentTag: string;
+    path: string;
+    args: string;
 
-    constructor (page: Page) {
+    constructor (page: Page, componentName: string, componentTag = 'data-test-id') {
         this.page = page;
+        this.componentName = componentName;
+        this.componentTag = componentTag;
+        this.path = '';
+        this.args = '';
+    }
+
+    async load (queries: Record<string, unknown> = {}) {
+        const pageUrl = buildUrl(this.componentName, this.composePath(queries), this.args);
+        await this.open(pageUrl);
+    }
+
+    async open (url: string) {
+        await this.page.goto(url, { waitUntil: 'networkidle' });
+        return this;
+    }
+
+    /**
+     * Composes the path passed in from the load function.
+     * If no query is passed in, returns the base component.
+     * Takes in objects with key/value pairs and flatmaps the queries into string values.
+     * Returns them as an interpolated string to be used in the URL.
+     *
+     * @param {string} queries The name of the attribute to retrieve.
+     * @returns args passed into URL seperated by ;
+     *
+     */
+    composePath (queries: Record<string, unknown>) {
+        if (!queries) {
+            return '';
+        }
+
+        const flattenQueries = (obj: Record<string, unknown>, prefix = ''): string[] => Object.entries(obj).flatMap(([key, value]) => {
+            const newKey = prefix ? `${prefix}.${key}` : key;
+
+            if (typeof value === 'object' && value !== null) {
+                return flattenQueries(value as Record<string, unknown>, newKey);
+            }
+            return `${newKey}:${value}`;
+        });
+
+        return `&args=${flattenQueries(queries).join(';')}`;
+    }
+
+    /**
+     * Listens for the given event. When it is emitted, the name of the event is pushed to an array on the `window` object.
+     */
+    async listenForEvent (eventName: string) {
+        await this.page.evaluate(() => {
+            window.__eventsArray = [];
```

Comment:
Just noticed that a side effect of this is that it will clear out any events that are already in the array. I don't think that's an issue if the setup is all done at the start of the test but just something to bear in mind.